File: 11_containerWithMostWater_1.py
#!/usr/bin/env python3

from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        current, left, right = 0, 0, len(height) - 1
        max_area = min(height[left], height[right]) * (right - left)
        while (left < right):
            if height[left] <= current:
                left += 1
            elif height[right] <= current:
                right -= 1
            current = min(height[left], height[right])
            max_area = max(max_area, current * (right - left))
        return max_area

File: 11_containerWithMostWater_1.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = Math.min(height[0], height[height.length - 1]) * (height.length - 1);
        int left = 0;
        int right = height.length - 1;
        int current = 0;
        while (left < right) {
            if (height[left] <= current) {
                left++;
                continue;
            }else if (height[right] <= current) {
                right--;
                continue;
            }
            current = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, current * (right - left));
        }
        return maxArea;
    }
}